src/pip/_internal/legacy_resolve.py

```diff
@@ -147,8 +147,8 @@ def __init__(
         self._discovered_dependencies = \
             defaultdict(list)  # type: DiscoveredDependencies
 
-    def resolve(self, requirement_set):
-        # type: (RequirementSet) -> None
+    def resolve(self, root_reqs, check_supported_wheels):
```

Comment:
Regarding the addition of `check_supported_wheels` to the API of the resolve call -- I think making `check_supported_wheels` an attribute of the resolver, would work better for us.

Yes, that'll create a Resolver->RequirementSet coupling, but given that we're now creating the RequirementSet in the resolver, it probably makes sense to do it that way. :)

Reply:
I don't disagree in general. Adding this to the resolver constructor adds some boilerplate (also requires updating `make_resolver`). Is it OK to defer this on the basis of that and:

1. This will probably be changing soon anyway when we sort out how we'll make new and old resolvers
2. As written it is a little easier to reason about where `check_supported_wheels` is used vs as a member vs a member, so it's easy to understand and change later

Reply:
Happy to defer this to a follow-up. :)